packages/flutter_driver/lib/src/driver/timeline.dart

```diff
@@ -124,11 +120,20 @@ class TimelineEvent {
 List<TimelineEvent> _parseEvents(Map<String, dynamic> json) {
   final List<dynamic> jsonEvents = json['traceEvents'] as List<dynamic>;
 
-  if (jsonEvents == null)
+  if (jsonEvents == null) {
     return null;
+  }
 
   // TODO(vegorov): use instance method version of castFrom when it is available.
-  return Iterable.castFrom<dynamic, Map<String, dynamic>>(jsonEvents)
-    .map<TimelineEvent>((Map<String, dynamic> eventJson) => TimelineEvent(eventJson))
-    .toList();
+  final List<TimelineEvent> timelineEvents =
+      Iterable.castFrom<dynamic, Map<String, dynamic>>(jsonEvents)
+          .map<TimelineEvent>(
+              (Map<String, dynamic> eventJson) => TimelineEvent(eventJson))
+          .toList();
+
+  timelineEvents.sort((TimelineEvent e1, TimelineEvent e2) {
+    return e1.timestampMicros.compareTo(e2.timestampMicros);
```

Comment:
I just tried running this on my favorite benchmark and got a null exception here. Apparently not all events have timestamps and so you need to do something special to compare events with null timestamps (null should be sorted as less than everything else, maybe?)

Reply:
It looks like all of the events in the timeline I was processing that had null timestamps were the "thread_name" events at the beginning of the log.